code/game/objects/structures.dm

```diff
@@ -64,26 +64,26 @@
 	if(C == user)
 		do_climb(user)
 
+/obj/structure/proc/density_check()
+	for(var/obj/O in range(0, src))
+		if(O.density == 1 && O != src && !istype(O, /obj/machinery/door/window)) //Ignores windoors, as those already block climbing, otherwise a windoor on the opposite side of a table would prevent climbing.
+			return O
+	for(var/turf/T in range(0, src))

```

Comment:
This doesn't really need to be a `for` loop given that it has a range of zero and only one turf can be on a coordinate space at a time. Just use `get_turf(src)` instead.